src/nvim/os/fs.c

```diff
@@ -773,6 +773,17 @@ int os_setperm(const char *const name, int perm)
   return (r == kLibuvSuccess ? OK : FAIL);
 }
 
+/// Set the permission of a file referred to by the open file
+/// descriptor, like fchmod(2).
+///
+/// @return 0 on success, or libuv error code on failure.
+int os_fsetperm(int fd, int perm)
```

Comment:
typically we would name this `os_ fchmod`.

Reply:
Yeah, in fact I had it like that to start with but thought it best to keep in line with `os_setperm` above. I'm happy either way

Reply:
oh, good point. Let's leave it as you have it.